### PR TITLE
Fix G38.2 in GCODE parser and probe reporting

### PR DESCRIPTION
**Pull request by StephanCoetzee**
Add sensor in grbl.

Fix in probe code in gcode parser.

Fix probe reporting to Motion.

Before merging, please test if we can a) detect missing magazines b) probe to a sensor on an axis c) Move to a magazine on the carousel using command line stepper.

### CODE REVIEW

**Comment by StephanCoetzee** [comment]
When I click on "Code" in Github, it says that there is -540 lines, but the diff doesn't show it. Has anyone else run into this scenario or am I doing something wrong?

File: probe.c [hunk 1]
```diff
@@ -159,11 +164,10 @@ void probe_move_to_sensor(float * target, float feed_rate, uint8_t invert_feed_r
   probe.isprobing = 1;
 
   sys.state = STATE_PROBING;
-  
-  if (!probe_loop())
-    return;
-
+ 
   uint8_t probe_fail;
+  probe_fail = !probe_loop();
```
Comment (Jeff-Ciesielski) [inline probe.c:178]: Wouldn't this just be overwritten by line 172?

File: gcode.c [hunk 2]
```diff
@@ -971,23 +971,20 @@ uint8_t gc_execute_line(char *line)
...
-            sensor = MAG_SENSOR; 
-          }
+        case MOTION_MODE_PROBE:          
           probe_move_to_sensor(gc_block.values.xyz, gc_state.feed_rate,
-          gc_state.modal.feed_rate, gc_block.values.n, sensor); 
+          gc_state.modal.feed_rate, gc_block.values.n, gc_block.values.p); 
           retval = STATUS_QUIET_OK;
           //block.values.xyz is updated inside probe_cycle, so the next line is correct
-        }
+
```
Comment (Jeff-Ciesielski) [inline gcode.c]: What's going on with this comment, where did it come from?

Comment (StephanCoetzee) [reply gcode.c]: Do you mean the long comment that starts with [G38 Errors]. That is from line ~820 in which the error checking is done (step 3 in the gcode parser). It should not be in here, I'll take it out. The `if (!axis_words)` statement should also not be here. The error checking is already done at this point.

Don't know how it got in here, my mistake. It is redundant.

File: probe.c (hunk 1 above)
Comment (StephanCoetzee) [reply probe.c:178]: Only if sensor == MAG_SENSOR. There was an existing check for probe_fail in the case of the MAG_SENSOR which depends on the PROBE_ACTIVE macro, which I kept.

File: gcode.c (hunk 2 above)
Comment (StephanCoetzee) [reply gcode.c]: Regarding the specifics of the comment - I didn't write that comment. 

It looks like the comment is related to the commented lines of code following it in which it checks if we are already at the position to which we try to probe to and if the probe is already triggered.

File: gcode.c [hunk 3]
```diff
@@ -89,6 +88,7 @@ static uint8_t gc_check_same_position(float *pos_a, float *pos_b)
 // coordinates, respectively.
 uint8_t gc_execute_line(char *line) 
 {
+  report_debug_message("received gcode");
```
Comment (Jeff-Ciesielski) [inline gcode.c]: If this prints all the time without a compiler flag being set, it should be removed.

Comment (StephanCoetzee) [reply gcode.c]: Thanks for catching this. It's just a debug message that slipped through. Will remove.

**Comment by mdicicco** [comment]
Is this ready to merge?  how does this relate to #76 ?

**Comment by Jeff-Ciesielski** [comment]
Running some tests on this, observed the following behavior: 

* Probing appears to work as expected with the command (at least in terms of being able to probe to an edge)
* Detecting missing magazine slats only works after you've homed, which is odd...
* I'm unable to detect a missing magazine while probing the carousel
* Probing is broken in motion. Gives `invalid gcode ID:39` (I think it isn't obeying the defaults for the `P` word). Not sure if the intent was backward compatibility, but this requires motion changes as-is (might be that there is a branch for motion as well).

**Comment by StephanCoetzee** [comment]
Regarding @Jeff-Ciesielski's comment. There is a branch in kiosk that I'm working on. So don't try probing with the existing kiosk branch. This PR shouldn't break anything except that probing in the current kiosk branch doesn't work. I try to merge new features in grbl first and then kiosk.

I made some changes and I'm ready to merge if you are.

**Pull request merged**